examples/cli-jobs/upload-local-data/confidential_io.py

```diff
@@ -0,0 +1,215 @@
+    )
+
+
+def config_global_rsa_key(
+    keyvault_url=None, rsa_key_name=None, managed_identity=None, lazy_init=True
+):
+    global _KEYVAULT_URL, _RSA_KEY_NAME, _MANAGED_IDENTITY
+    _KEYVAULT_URL = (
+        _KEYVAULT_URL
+        or keyvault_url
+        or os.environ.get("CONFIDENTIALITY_KEYVAULT", None)
+    )
+    _RSA_KEY_NAME = (
+        _RSA_KEY_NAME
+        or rsa_key_name
+        or os.environ.get("CONFIDENTIALITY_KEY_NAME", None)
+    )
+    _MANAGED_IDENTITY = (
+        _MANAGED_IDENTITY
+        or managed_identity
+        or os.environ.get("DEFAULT_IDENTITY_CLIENT_ID", None)
+    )
+
+    if not lazy_init:
+        # initialize right away
+        get_rsa_client()
+
+
+def get_rsa_client(keyvault_url=None, rsa_key_name=None, managed_identity=None):
+    """Get crypto client from Keyvault for RSA key."""
+    global _RSA_CRYPTO_CLIENT, _KEYVAULT_URL, _RSA_KEY_NAME, _MANAGED_IDENTITY
+
+    # let's not recreate a new client
+    if _RSA_CRYPTO_CLIENT is not None:
+        logging.getLogger(__name__).info(
+            "CryptographyClient already initialized, returning existing instance."
+        )
+        return _RSA_CRYPTO_CLIENT
+
+    # if not done already, capture references from env vars
+    config_global_rsa_key(keyvault_url=keyvault_url, rsa_key_name=rsa_key_name)
+
+    if _KEYVAULT_URL is None:
+        raise ValueError(
+            "To connect to the keyvault you need to provide its url, call config_global_rsa_key(keyvault_url, rsa_key_name) first or call config_global_rsa_key() with keyvault_url not None"
+        )
+    if _RSA_KEY_NAME is None:
+        raise ValueError(
+            "To connect to the keyvault you need to provide a key name, call config_global_rsa_key(keyvault_url, rsa_key_name) first or call config_global_rsa_key() with rsa_key_name not None"
+        )
+
+    # get credentials
+    if _MANAGED_IDENTITY:
+        # running in AzureML with a compute identity assigned
+        logging.getLogger(__name__).info(
+            f"Using ManagedIdentityCredential with client_id={_MANAGED_IDENTITY} to connect to keyvault"
+        )
+        credential = ManagedIdentityCredential(client_id=_MANAGED_IDENTITY)
+    else:
+        logging.getLogger(__name__).info(
+            f"Using DefaultAzureCredential to connect to keyvault"
+        )
+        credential = DefaultAzureCredential()
+
+    # get a client to the keyvault
+    logging.getLogger(__name__).info(f"Connecting to keyvault {_KEYVAULT_URL}...")
+    key_client = KeyClient(vault_url=_KEYVAULT_URL, credential=credential)
+
+    # create crypto client to help with encryption
+    logging.getLogger(__name__).info(
+        f"Obtaining CryptographyClient for key {_RSA_KEY_NAME}..."
+    )
+    _RSA_CRYPTO_CLIENT = key_client.get_cryptography_client(key_name=_RSA_KEY_NAME)
+
+    return _RSA_CRYPTO_CLIENT
+
+
+class read_encrypted:
```

Comment:
I was hesitant at first, I wanted the "user side" to use something like `with read_encrypted() as f` rather than `with EncryptionHandler() as f`. But that could be changed.

Reply:
I finally opted for camel case...